embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/opentelemetry/OpenTelemetryConfiguration.kt

```diff
@@ -9,15 +10,34 @@ import io.embrace.android.embracesdk.internal.spans.EmbraceSpanProcessor
 import io.embrace.android.embracesdk.internal.spans.SpanSink
 import io.opentelemetry.sdk.logs.LogRecordProcessor
 import io.opentelemetry.sdk.logs.export.LogRecordExporter
+import io.opentelemetry.sdk.resources.Resource
 import io.opentelemetry.sdk.trace.SpanProcessor
 import io.opentelemetry.sdk.trace.export.SpanExporter
+import io.opentelemetry.semconv.incubating.AndroidIncubatingAttributes
+import io.opentelemetry.semconv.incubating.DeviceIncubatingAttributes
+import io.opentelemetry.semconv.incubating.OsIncubatingAttributes
+import io.opentelemetry.semconv.incubating.ServiceIncubatingAttributes
 
 internal class OpenTelemetryConfiguration(
     spanSink: SpanSink,
-    logSink: LogSink
+    logSink: LogSink,
+    systemInfo: SystemInfo
 ) {
     val serviceName = BuildConfig.LIBRARY_PACKAGE_NAME
     val serviceVersion = BuildConfig.VERSION_NAME
+    val resource: Resource = Resource.getDefault().toBuilder()
```

Comment:
One subtle difference between this and how we populate our payloads is how `os.type` and `os.name` are defined in OTel. For this, I'm following the [guide lines here](https://opentelemetry.io/docs/specs/semconv/resource/os/)

OTel defines Android's OS type as "linux" because that's what the kernel is, so I respect that in `os.type` - however, we use "Android OS" in our payload for the attribute `os_type` right now, which is a bit confusing. If anything, it should just be `android`, but we can resolve that separately.

To identify as Android, we use `os.name`, which we don't use in our payload, so that's fine.